### ISSUE TITLE
keras version of EnvNet

### ISSUE THREAD

**Issue by NemoCpp** [mil-tokyo/bc_learning_sound#7]
Hi Yuji Tokozume and other authors,
I am not familiar with chainer module, so I rewrite the EnvNet according to your paper and your chainer version code. here is the code:

```
import numpy as np
np.random.seed(1)
import tensorflow as tf
tf.set_random_seed(1)
import keras
keras.backend.clear_session()
from keras.models import Sequential
from keras.layers import Activation
from keras.layers import Reshape
from keras.layers import Flatten
from keras.layers import Convolution2D, MaxPooling2D, Dropout, Conv1D, MaxPooling1D
from keras.layers.normalization import BatchNormalization
from keras.layers.core import Dense
from keras.layers.core import Permute

def EnvNet(input_length=24000, num_classes=50):

    model = Sequential()

    # section 1, conv1
    model.add(
        Conv1D(
            input_shape=(input_length, 1),
            filters=40, 
            kernel_size=8,
            strides=1,
            padding="same",
        )
    )

    model.add(BatchNormalization())
    model.add(Activation('relu'))
    
    # section 2, conv2
    model.add(
        Conv1D(
            filters=40, 
            kernel_size=8,
            strides=1,
            padding="same",
        )
    )

    model.add(BatchNormalization())
    model.add(Activation('relu'))
    
    # section 3, max_pooling_2d
    model.add(MaxPooling1D(pool_size=(160)))
    
    # section 4, swapaxes
    model.add(Permute((2,1)))
    model.add(Reshape((model.output_shape[1], model.output_shape[2], 1)))

    # section 5, conv3
    model.add(
        Convolution2D(
            filters=50, 
            kernel_size=(8,13),
            strides=1,
            data_format="channels_last",
        )
    )
    model.add(BatchNormalization())
    model.add(Activation('relu'))
    
    # section 6, max_pooling_2d
    model.add(MaxPooling2D(pool_size=(3, 3)))

    # section 7, conv4
    model.add(
        Convolution2D(
            filters=50, 
            kernel_size=(1,5),
            strides=1,
        )
    )
    model.add(BatchNormalization())
    model.add(Activation('relu'))
    
    # section 8, max_pooling_2d
    model.add(MaxPooling2D(pool_size=(1, 3)))

    # section 9, F.dropout(F.relu(self.fc5(h)), rain=self.train)
    model.add(Flatten())
    model.add(Dense(4096))
    model.add(Activation('relu'))
    model.add(Dropout(0.5))

    # section 10, F.dropout(F.relu(self.fc6(h)), rain=self.train)
    model.add(Dense(4096))
    model.add(Activation('relu'))
    model.add(Dropout(0.5))

    # section 11, fc7(h)
    model.add(Dense(num_classes))

    return model
```
and below is model summary

_________________________________________________________________
Layer (type)                 Output Shape              Param #
=================================================================
conv1d_28 (Conv1D)           (None, 24000, 40)         360
_________________________________________________________________
batch_normalization_59 (Batc (None, 24000, 40)         160
_________________________________________________________________
activation_63 (Activation)   (None, 24000, 40)         0
_________________________________________________________________
conv1d_29 (Conv1D)           (None, 24000, 40)         12840
_________________________________________________________________
batch_normalization_60 (Batc (None, 24000, 40)         160
_________________________________________________________________
activation_64 (Activation)   (None, 24000, 40)         0
_________________________________________________________________
max_pooling1d_13 (MaxPooling (None, 150, 40)           0
_________________________________________________________________
permute_18 (Permute)         (None, 40, 150)           0
_________________________________________________________________
reshape_11 (Reshape)         (None, 40, 150, 1)        0
_________________________________________________________________
conv2d_39 (Conv2D)           (None, 33, 138, 50)       5250
_________________________________________________________________
batch_normalization_61 (Batc (None, 33, 138, 50)       200
_________________________________________________________________
activation_65 (Activation)   (None, 33, 138, 50)       0
_________________________________________________________________
max_pooling2d_26 (MaxPooling (None, 11, 46, 50)        0
_________________________________________________________________
conv2d_40 (Conv2D)           (None, 11, 42, 50)        12550
_________________________________________________________________
batch_normalization_62 (Batc (None, 11, 42, 50)        200
_________________________________________________________________
activation_66 (Activation)   (None, 11, 42, 50)        0
_________________________________________________________________
max_pooling2d_27 (MaxPooling (None, 11, 14, 50)        0
_________________________________________________________________
flatten_2 (Flatten)          (None, 7700)              0
_________________________________________________________________
dense_7 (Dense)              (None, 4096)              31543296
_________________________________________________________________
activation_67 (Activation)   (None, 4096)              0
_________________________________________________________________
dropout_5 (Dropout)          (None, 4096)              0
_________________________________________________________________
dense_8 (Dense)              (None, 4096)              16781312
_________________________________________________________________
activation_68 (Activation)   (None, 4096)              0
_________________________________________________________________
dropout_6 (Dropout)          (None, 4096)              0
_________________________________________________________________
dense_9 (Dense)              (None, 50)                204850
=================================================================
Total params: 48,561,178
Trainable params: 48,560,818
Non-trainable params: 360
_________________________________________________________________

could you please help me to check whether I am right or wrong ?
